pageserver/src/walingest.rs

```diff
@@ -1530,14 +1590,27 @@ impl WalIngest {
         xlrec: &XlMultiXactTruncate,
         ctx: &RequestContext,
     ) -> Result<()> {
-        self.checkpoint.oldestMulti = xlrec.end_trunc_off;
-        self.checkpoint.oldestMultiDB = xlrec.oldest_multi_db;
-        self.checkpoint_modified = true;
+        let (modified, maxsegment, startsegment, endsegment) =
+            enum_pgversion_dispatch!(&mut self.checkpoint, CheckpointVersion, cp, {
+                let modified = cp.oldestMultiDB != xlrec.oldest_multi_db
+                    || cp.oldestMulti != xlrec.end_trunc_off;
+                cp.oldestMulti = xlrec.end_trunc_off;
+                cp.oldestMultiDB = xlrec.oldest_multi_db;
+                let maxsegment: i32 = pgv::nonrelfile_utils::mx_offset_to_member_segment(
+                    pg_constants::MAX_MULTIXACT_OFFSET,
+                );
+                let startsegment: i32 =
+                    pgv::nonrelfile_utils::mx_offset_to_member_segment(xlrec.start_trunc_memb);
+                let endsegment: i32 =
+                    pgv::nonrelfile_utils::mx_offset_to_member_segment(xlrec.end_trunc_memb);
+                (modified, maxsegment, startsegment, endsegment)
+            });
+
+        if modified {
+            self.checkpoint_modified = true;
+        }
```

Comment:
This is

1. drive-by micro-optimization to improve precision on checkpoint_modified
2. dispatching to code that is actually not postgres-versioned, so, we can avoid the dispatch.